docs/configuration.md

```diff
@@ -0,0 +1,94 @@
+# Configuration
+
+See the default config [here](../config/default.yml)
+
+---
+### allowUnused
+List of modules that are allowed to be unused.
+Passed to `string.match`
+
+Please create an [issue](https://github.com/charlierudolph/dependency-lint/issues)
+anytime you need to use this
+
+Example:
+```yml
+allowUnused:
+  - mocha
+```
+
+---
+### devFilePatterns
+Files used only for development.
+Uses [minimatch](https://github.com/isaacs/minimatch)
```

Comment:
I always wondered whether we need to mention the particular library that parses this here. I think everybody understands glob patterns pretty well, especially with the example.